backend/src/impl/internal_models/system_model.py

```diff
@@ -132,18 +137,29 @@ def save_system_output(
         if properties.get("system_output"):
             # delete previously saved system_output
             get_storage().delete([properties["system_output"]])
-        sample_list = [general_to_dict(v) for v in system_output.samples]
         blob_name = f"{self.system_id}/{self._SYSTEM_OUTPUT_CONST}"
         get_storage().compress_and_upload(
             blob_name,
-            json.dumps(sample_list),
+            json.dumps(system_output.samples),
+        )
+        serializer = PrimitiveSerializer()
+        system_output_metadata = dataclasses.asdict(system_output.metadata)
+        system_output_metadata["custom_features"] = serializer.serialize(
+            system_output.metadata.custom_features
         )
+        # TODO(lyuyang): This related to a bug in the SDK: custom_analyses aren't
```

Comment:
Is there an issue for this in the SDK repo? If not, could you create one? If so, could you link to it?

Reply:
I have created an https://github.com/neulab/ExplainaBoard/issues/596. I'll also submit a PR to fix it.